client/elements/text/sc-segmented-text.html

```diff
@@ -259,66 +272,99 @@
           }
 
           _updateToolbar() {
-              this.dispatch('changeSuttaMetaText', this._computeMeta());
               this.dispatch('changeToolbarTitle',
-                  `${this.translatedTitle ? this.translatedTitle : this.rootTitle} – ${this.translationAuthor}`);
+                  `${this.translatedTitle || this.rootTitle} – ${this.translationAuthor || this.rootAuthor}`);
           }
 
           // returns the meta-data from the loaded sutta text
           _computeMeta() {
-              let matches = this.translatedSutta.text.match(/<aside id="metaarea">((.|\n)*)<\/aside>/);
-              if (!matches) {
-                  matches = this.rootSutta.text.match(/<aside id="metaarea">((.|\n)*)<\/aside>/);
+              if (this.translatedSutta) {
+                  return this.translatedSutta.author_blurb[this.translationLang] || 'No metadata available.';
+              } else {
+                  return 'No metadata available.';
               }
-              let metaText;
-              try {
-                  if (matches && matches.length > 0) {
-                      metaText = matches[1];
-                  } else {
-                      metaText = '';
-                  }
-              } catch (e) {
-                  console.error(e);
+          }
+
+          // if the base html and the pali texts have been fully loaded, the setting from the
+          // settings-menu for the relevant view is implemented by adding relevant class
+          // or by adding a paper-tooltip and populating this with the pali text in the chosen script.
+          _setViewOptions() {
+              if (!this.isSegmented || !this.markup || !this.translatedSutta) {
+                  return;
               }
-              return metaText;
+              this._resetViewOptions();
+              const textContent = this.$.segmented_text_content;
+              switch (this.chosenTextView) {
+                  case 'sidebyside':
+                      textContent.classList.add('side-by-side');
+                      this._addSecondaryText();
+                      break;
+                  case 'linebyline':
+                      textContent.classList.add('line-by-line');
+                      this._addSecondaryText();
+                      break;
+                  case 'popup':
+                      Array.from(textContent.querySelectorAll('.translated-text')).forEach(item =>
```

Comment:
It is a better practice to have up to 2 lines in switch/case (extract to function)